src/syft.js

```diff
@@ -7,159 +7,79 @@ import {
-      if (this.lastUnfinishedOperation === opsLength)
-        throw new Error(PLAN_ALREADY_COMPLETED(this.plan.name, this.plan.id));
-
-      // For each argument supplied, store them in this.objects
-      data.forEach((datum, i) => {
-        this.objects[inputPlaceholders[i].id] = datum;
-      });
-
-      // Add state tensors to objects
-      if (this.plan.state && this.plan.state.tensors) {
-        this.plan.state.tensors.forEach(tensor => {
-          this.objects[tensor.id] = tensor;
-        });
-      }
-
-      let finished = true;
-
-      // Execute the plan
-      for (let i = this.lastUnfinishedOperation; i < opsLength; i++) {
-        // The current operation
-        const currentOp = this.plan.operations[i];
-
-        // The result of the current operation
-        const result = currentOp.execute(this.objects, this.logger);
-
-        // Place the result of the current operation into this.objects at the 0th item in returnIds
-        if (result) {
-          if (currentOp.returnIds.length > 0) {
-            this.objects[currentOp.returnIds[0]] = result;
-          } else if (currentOp.returnPlaceholders.length > 0) {
-            this.objects[currentOp.returnPlaceholders[0].id] = result;
-          }
-        } else {
-          finished = false;
-          this.lastUnfinishedOperation = i;
-
-          break;
-        }
-      }
-
-      if (finished) {
-        // Set the lastUnfinishedOperation as the number of operations (meaning, we've already executed the plan successfully)
-        this.lastUnfinishedOperation = opsLength;
-
-        // Resolve all of the requested resultId's as specific by the plan
-        const resolvedResultingTensors = [];
-        const outputPlaceholders = this.plan.getOutputPlaceholders();
-        outputPlaceholders.forEach(placeholder => {
-          resolvedResultingTensors.push({
-            id: placeholder.id,
-            value: this.objects[placeholder.id]
-          });
-        });
-
-        // Return them to the worker
-        resolve(resolvedResultingTensors);
-      } else {
-        // If the plan wasn't finished, notify the worker that they may try again once they have the appropriate information
-        reject(
-          'There is not enough information to execute this plan, but we have saved your progress!'
-        );
-      }
-    });
+  /**
+   * Load the model into worker's models registry
+   * @param requestKey
+   * @param modelId
+   * @returns {Promise<SyftModel>}
+   */
+  async loadModel({ requestKey, modelId }) {
+    const modelData = await this.gridClient.getModel(
+      this.id,
+      requestKey,
+      modelId
+    );
+    const model = new SyftModel({ worker: this, modelData });
+    this.models.set(modelId, model);
+    return model;
   }
 
-  /* ----- EVENT HANDLERS ----- */
-
-  onSocketStatus(func) {
-    this.observer.subscribe(SOCKET_STATUS, func);
+  getConnectionInfo() {
```

Comment:
This should be done in the `newJob` function above.

Reply:
I moved get connection speed into GridApiClient.
It is then used in job.start() to get connection speed to send it to cycle-request.